src/main/kotlin/no/nav/familie/ba/sak/kjerne/vedtak/vedtaksperiode/VedtaksperiodeUtil.kt

```diff
@@ -148,28 +148,49 @@ fun identifiserReduksjonsperioder(
     vedtak: Vedtak,
     opphørsperioder: List<VedtaksperiodeMedBegrunnelser>
 ): List<VedtaksperiodeMedBegrunnelser> {
-    val forrigeSegmenter = forrigeAndelerTilkjentYtelse.lagVertikaleSegmenter().keys
-    val nåværendeSegmenter = andelerTilkjentYtelse.lagVertikaleSegmenter().keys
-    val segmenter = forrigeSegmenter.filterNot {
-        nåværendeSegmenter.any { segment -> it.fom == segment.fom && it.tom == segment.tom && it.value == segment.value }
+    val forrigeSegmenter = forrigeAndelerTilkjentYtelse.lagVertikaleSegmenter()
+    val nåværendeSegmenter = andelerTilkjentYtelse.lagVertikaleSegmenter()
+    val segmenter = forrigeSegmenter.filterNot { (forrigeSegment, _) ->
+        nåværendeSegmenter.any { (nyttSegment, _) ->
+            forrigeSegment.fom == nyttSegment.fom &&
+                forrigeSegment.tom == nyttSegment.tom &&
+                forrigeSegment.value == nyttSegment.value
+        }
     }
-    val reduksjonsperioder = mutableListOf<VedtaksperiodeMedBegrunnelser>()
-    segmenter.filter { nåværendeSegmenter.any { segment -> segment.overlapper(it) } }
-        .forEach {
-            val overlappendePerioder =
-                nåværendeSegmenter.filter { nåSegment -> nåSegment.overlapper(it) && nåSegment.value < it.value }
-            overlappendePerioder.forEach { overlappendePeriode ->
-                reduksjonsperioder.add(
+    val reduksjonsperioder =
+        segmenter.filter { (forrigeSegment, _) ->
+            nåværendeSegmenter.any { (nyttSegment, _) ->
+                nyttSegment.overlapper(
+                    forrigeSegment
+                )
+            }
+        }.toList()
+            .fold(emptyList<VedtaksperiodeMedBegrunnelser>()) { acc, (gammeltSegment, andeltilkjenYtelserForSegment) ->
+                val overlappendePerioder =
+                    nåværendeSegmenter
+                        .filter { (nåSegment, nåAndelTilkjentYtelserForSegment) ->
+                            nåSegment.overlapper(gammeltSegment) && andeltilkjenYtelserForSegment.any { andelerTilkjentYtelse ->
+                                nåAndelTilkjentYtelserForSegment.any {
+                                    it.aktør.aktørId == andelerTilkjentYtelse.aktør.aktørId &&
+                                        it.sats < andelerTilkjentYtelse.sats &&
+                                        // Perioder med delt bosted eller endret utbetaling er ikke vurdert som reduksjonsperioder
+                                        (!it.erDeltBosted() && it.endretUtbetalingAndeler.isEmpty())
+                                } || nåAndelTilkjentYtelserForSegment.all {
+                                    // Når et av barna mister utbetaling på et segment i behandling
+                                    it.aktør.aktørId != andelerTilkjentYtelse.aktør.aktørId
+                                }
```

Comment:
Ref det Henning snakket om i møtet. Det er kanskje nok å se om en andel har forsvunnet?

```suggestion
nåAndelTilkjentYtelserForSegment.all {
                                    // Når et av barna mister utbetaling på et segment i behandling
                                    it.aktør.aktørId != andelerTilkjentYtelse.aktør.aktørId
                                }
```

Reply:
Yes, her kunne/burde man også gjenbrukt LocalDateTimeline logikk som blir brukt for å finne endringstidspunkt. Der finner man jo også reduksjoner automatisk :)